fem/bilininteg.hpp

```diff
@@ -2995,6 +2997,26 @@ class ElasticityIntegrator : public BilinearFormIntegrator
    Vector divshape;
 #endif
 
+   // PA extension
+   std::shared_ptr<QuadratureFunction> lambda_quad, mu_quad;
+   std::shared_ptr<QuadratureFunction> q_vec;
```

Comment:
You might want to use `CoefficientVector` instead of `QuadratureFunction`. This has the potential to allow for some compression (e.g. storing a constant coefficient as only a single value rather than a full quadrature function).

Reply:
That would be nice to reduce memory but I don't think `CoefficientVector` can be used on GPUs.

Reply:
`CoefficientVector` definitely supports GPUs: it's derived from `Vector`, you can call `Vector::Read` as usual to get the device pointer. (You can see some examples of its use in the PA diffusion or mass integrators)

Reply:
I thought you were talking about `VectorCoefficient` and didn't realize there was a separate `CoefficientVector`.

Reply:
I turned `lambda_quad` and `mu_quad` into `CoefficientVectors` s and kept `q_vec` as a `QuadratureFunction` since just scratch space.

It seems like utilizing compression requires modifying the kernel for the different compression levels. I didn't want to do that yet for the constant case because I don't have a test for that currently, but at least it will be easy to add in the future now. I verify that it doesn't use the constant structure before calling the kernels like the diffusion one.